docs/zkapps/advanced-snarkyjs/actions-and-reducer.mdx

```diff
@@ -99,9 +99,63 @@ this.reducer.getActions({ fromActionsHash?: Field, endActionsHash?: Field }): A[
 
 Right now, `getActions` is available for [testing with `LocalBlockchain`](/zkapps/how-to-test-a-zkapp).
 
+### Fetching Actions from an Archive Node
+
+Since Mina nodes do not store historical network information and actions are not kept on-chain, emitted actions are only preserved in the Archive Node. Consequently, a zkApp must fetch its previously emitted actions on the Mina network from an Archive Node.
+
+As the Archive Node is essentially a database, it does not offer a user-friendly API that SnarkyJS can readily utilize to fetch actions. To address this issue, a GraphQL server was developed to retrieve network information such as actions and actions from the Archive Node. SnarkyJS will use this GraphQL server to fetch actions from the Archive Node.
```

Comment:
```suggestion
As the Archive Node is essentially a database, it does not offer a user-friendly API that SnarkyJS can readily utilize to fetch actions. To address this issue, a GraphQL server was developed to retrieve network information such as actions and actions from the Archive Node. SnarkyJS uses this GraphQL server to fetch actions from the Archive Node.
```

Reply:
even when future events are mentioned, style guidance is to use present tense https://developers.google.com/style/tense please apply throughout

Reply:
hit up the present tense throughout, thanks!

Reply:
Note "actions and actions". Might be intended as "events as actions"?

Reply:
Good catch

Reply:
I'm going to suggest a bigger change here 1.) to avoid confusion over database vs API, and 2.) to highlight the ability to store data using any datastore, given past experience where community wrongly read into old docs on this specific aspect, and we rewrite the docs to be more clear that any data store can be used.

> The Mina archive node offers the ability for operators to store data using any data store of their choice, however, it does not currently expose a built-in API to query this data publicly. To address this, the [Mina Archive Node API](https://github.com/o1-labs/archive-node-api) was developed and can be run by archive node operators to provide a GraphQL API from which zkApp smart contracts can retrieve events and actions.